### PR TITLE
feat: API to login and get user username, name and email 

### PR DESCRIPTION
**Pull request by stephanegigandet**
This is to address #7361 so that the Flutter app can get the username and name of an user who authenticated with an email address.

but it could also be a step to have a more uniform handling of errors in the API (specifically errors related to users not identified or bad logins and passwords supplied).

I'm proposing that we extend the /cgi/auth.pl API which currently:

- checks if an user is identified (either through a session cookie on the *.openfoodfacts.org domain, or by passing user_id and password)
- returns a 200 status code if the user is logged in, or a 403 otherwise

The extension would be to add a JSON body:
- when the user is logged in:

```json
{
    "status": 1,
    "status_verbose": "user signed-in",
    "user": {
        "email": "my email",
        "name": "Stéphane Gigandet"
    },
    "user_id": "test2"
}

```
- when the user is not logged in:
```json
{
    "status": 0,
    "status_verbose": "user not signed-in"
}
```

With the proposed code, this behaviour would only apply to /cgi/auth.pl

But we could decide to extend it to all other API calls (e.g. product edit). If invalid userid / password are supplied, we return a JSON body to explain why (instead of returning an HTML content today).

I think we could keep the existing status: 0 that we have in most APIs I think (get and search product, product edit, auth), and maybe add an "error" id + and "error_message" in the language requested.

cc @VaiTon @M123-dev @monsieurtanuki @g123k

### CODE REVIEW

**Comment by M123-dev** [comment]
I love the response body, as long as the Status Codes don't change and so break the older versions I really like this feature.

Same goes for the other api calls. It will be a relieve if we can just foreward a consistent error to the sdk user instead of manually parsing the html and extracting errors out of it. Especially if the messages are translated.

**Comment by alexgarel** [comment]
@stephanegigandet would you add it to the api.yml file ?

File: cgi/auth.pl [hunk 1]
```diff
@@ -63,5 +85,12 @@
...
+print header(-status => $status, -type => 'application/json', -charset => 'utf-8');
+print $json;
+
 $r->rflush;
-$r->status($status);
+
+# Setting the status makes mod_perl append a default error to the body
+# $r->status($status);
+# Send 200 instead.
+$r->status(200);
```
Comment (alexgarel) [inline cgi/auth.pl:96]: Hum ain't it too big a change for third party app ? (javascript app etc.).

Maybe we should return json only if the user add a json=1 parameter ? (and in this case return a 200).

Comment (stephanegigandet) [reply cgi/auth.pl:96]: The only apps that use /cgi/auth.pl are Folksonomy Engine and Hunger Games I believe (it's not documented anywhere). And I think they only look at the status code, as we were only returning a status code, with an empty body.

Comment (stephanegigandet) [reply cgi/auth.pl:96]: Note that this $r->status(200) actually does not change the true return status that is sent to the client, which is set in the header above.

image

**Comment by stephanegigandet** [comment]
> @stephanegigandet would you add it to the api.yml file ?

Yes, I wanted to discuss it a bit before writing the documentation.

**Comment by hangy** [comment]
What's the integration story for Ory's Hydra & Kratos? It seems like there might be some feature overlap with OIDC profile claims.

**Comment by alexgarel** [comment]
> What's the integration story for Ory's Hydra & Kratos? It seems like there might be some feature overlap with OIDC profile claims.

We might take info from Ory Kratos when it's integrated, but it will take some times.

**Review by alexgarel — APPROVED** [review]
LGTM.

Maybe, if you can, document before merging :-) (or after then)

**Comment by monsieurtanuki** [comment]
Cool feature! Ready to implement it on off-dart when it's merged here.

**Pull request merged**